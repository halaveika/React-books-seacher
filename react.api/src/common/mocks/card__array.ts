export const mockCardArr = [
  {
    id: 'B096D6BCTF',
    title: 'The Unsinkable Greta James The Unsinkable Greta James The Unsinkable Greta James',
    author: 'Jennifer E. Smith',
    vote_average: 4.3,
    poster: '41k5ka3zAML.jpg',
    overview:
      'Right after the sudden death of her mother—her first and most devoted fan—and just before the launch of her high-stakes sophomore album, Greta James falls apart on stage. The footage quickly goes viral and she stops playing, her career suddenly in jeopardy—the kind of jeopardy her father, Conrad, has always predicted; the kind he warned her about when he urged her to make more practical choices with her life.',
    price: 15.59,
  },
  {
    id: 'B096HF9RJ4',
    title: 'All My Rage',
    author: 'Sabaa Tahir',
    vote_average: 4.7,
    poster: '41EcpwOA81L.jpg',
    overview:
      'Misbah is a dreamer and storyteller, newly married to Toufiq in an arranged match. After their young life is shaken by tragedy, they come to the United States and open the Clouds Rest Inn Motel, hoping for a new start. ',
    price: 13.19,
  },
  {
    id: 'B092T8G139',
    title: 'Never Simple',
    author: 'Liz Scheier',
    vote_average: 4.3,
    poster: '41XpGHtDv0L.jpg',
    overview:
      'Scheier’s mother Judith was a news junkie, a hilarious storyteller, a fast-talking charmer you couldn’t look away from, a single mother whose devotion crossed the line into obsession, and—when in the grips of the mental illness that plagued every day of her life—a violent and abusive liar whose hold on reality was shaky at best. On an uneventful afternoon when Scheier was eighteen, her mother sauntered into the room to tell her two important things: one, she had been married for most of Scheier’s life to a man she’d never heard of, and two, the man she’d told Scheier was her father was entirely fictional.',
    price: 16.79,
  },
  {
    id: 'B0927CQNWV',
    title: 'The Chase',
    author: 'Candice Fox',
    vote_average: 4.2,
    poster: '41fDgD00h8L.jpg',
    overview:
      'With that, the largest manhunt in United States history is on. In response to a hostage situation, more than 600 inmates from the Pronghorn Correctional Facility, including everyone on Death Row, are released into the Nevada Desert. Criminals considered the worst of the worst, monsters with dark, violent pasts, are getting farther away by the second.',
    price: 14.99,
  },
  {
    id: 'B096XGDWSX',
    title: 'In Love',
    author: 'Amy Bloom',
    vote_average: 4.6,
    poster: '41hcaivkDML.jpg',
    overview:
      'Amy Bloom began to notice changes in her husband, Brian: He retired early from a new job he loved; he withdrew from close friendships; he talked mostly about the past. Suddenly, it seemed there was a glass wall between them, and their long walks and talks stopped. Their world was altered forever when an MRI confirmed what they could no longer ignore: Brian had Alzheimer’s disease.',
    price: 16.79,
  },
  {
    id: 'B097B2DXS8',
    title: 'Disorientation',
    author: 'Elaine Hsieh Chou',
    vote_average: 4,
    poster: '41xRDAKyOWL.jpg',
    overview:
      'Twenty-nine-year-old PhD student Ingrid Yang is desperate to finish her dissertation on the late canonical poet Xiao-Wen Chou and never read about “Chinese-y” things again. But after years of grueling research, all she has to show for her efforts are junk food addiction and stomach pain. When she accidentally stumbles upon a curious note in the Chou archives one afternoon, she convinces herself it’s her ticket out of academic hell.',
    price: 17.99,
  },
  {
    id: 'B09GXL26SV',
    title: 'The Atlas Six',
    author: 'Olivie Blake',
    vote_average: 4.4,
    poster: '51LzpTvAR8L.jpg',
    overview:
      'The Alexandrian Society, caretakers of lost knowledge from the greatest civilizations of antiquity, are the foremost secret society of magical academicians in the world. Those who earn a place among the Alexandrians will secure a life of wealth, power, and prestige beyond their wildest dreams, and each decade, only the six most uniquely talented magicians are selected to be considered for initiation.',
    price: 9.59,
  },
  {
    id: 'B096DN56XX',
    title: 'The Emergency: A Year of Healing and Heartbreak',
    author: 'Thomas Fisher',
    vote_average: 2.5,
    poster: '41eas-fn+GL.jpg',
    overview:
      'As an emergency room doctor working on the rapid evaluation unit, Dr. Thomas Fisher has about three minutes to spend with the patients who come into the South Side of Chicago ward where he works before directing them to the next stage of their care. Bleeding: three minutes. Untreated wound that becomes life-threatening: three minutes. Kidney failure: three minutes. He examines his patients inside and out, touches their bodies, comforts and consoles them, and holds their hands on what is often the worst day of their lives. Like them, he grew up on the South Side; this is his community and he grinds day in and day out to heal them.',
    price: 16.79,
  },
  {
    id: 'B092T8WM4Y',
    title: 'Mecca',
    author: 'Susan Straight',
    vote_average: 4.2,
    poster: '410MzVi+ngS.jpg',
    overview:
      'Johnny Frías has California in his blood. A descendant of the state’s Indigenous people and Mexican settlers, he has Southern California’s forgotten towns and canyons in his soul. He spends his days as a highway patrolman pulling over speeders, ignoring their racist insults, and pushing past the trauma of his rookie year, when he killed a man assaulting a young woman named Bunny, who ran from the scene, leaving Johnny without a witness. But like the Santa Ana winds that every year bring the risk of fire, Johnny’s moment of action twenty years ago sparked a slow-burning chain of connections that unites a vibrant, complex cast of characters in ways they never see coming.',
    price: 17.99,
  },
  {
    id: 'B0927B1P8L',
    title: 'The Kaiju Preservation Society',
    author: 'John Scalzi',
    vote_average: 4.6,
    poster: '51WcP+w2TDS.jpg',
    overview:
      'When COVID-19 sweeps through New York City, Jamie Gray is stuck as a dead-end driver for food delivery apps. That is, until Jamie makes a delivery to an old acquaintance, Tom, who works at what he calls “an animal rights organization.” Tom’s team needs a last-minute grunt to handle things on their next field visit. Jamie, eager to do anything, immediately signs on.',
    price: 16.79,
  },
];
