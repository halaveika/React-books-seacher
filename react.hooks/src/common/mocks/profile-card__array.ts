export const mockCardArr = [
  {
    id: 'test1-id',
    firstname: 'firstname',
    lastname: 'lastname',
    zipcode: '22555',
    birthday: '2011-01-01',
    gender: 'test1-gender',
    avatar: 'test1-avatar',
    city: 'test1-city',
    gifts: [],
  },
  {
    id: 'test2-id',
    firstname: 'test2-firstname',
    lastname: 'test2-firstname',
    zipcode: '22555',
    birthday: '2011-01-01',
    gender: 'test2-gender',
    avatar: 'test2-avatar',
    city: 'test2-city',
    gifts: ['test2-gift1', 'test2-gift2'],
  },
  {
    id: 'test3-id',
    firstname: 'test3-firstname',
    lastname: 'test3-firstname',
    zipcode: '22555',
    birthday: '2011-01-01',
    gender: 'test3-gender',
    avatar: 'test3-avatar',
    city: 'test3-city',
    gifts: ['test3-gift1', 'test3-gift2'],
  },
];
