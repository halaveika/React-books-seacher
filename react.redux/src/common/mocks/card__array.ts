export const mockCardArr = [
  {
    id: 'B096D6BCTF',
    title: 'The Unsinkable Greta James The Unsinkable Greta James The Unsinkable Greta James',
    authors: ['Jennifer E. Smith'],
    averageRating: 4.3,
    description:
      'Right after the sudden death of her mother—her first and most devoted fan—and just before the launch of her high-stakes sophomore album, Greta James falls apart on stage. The footage quickly goes viral and she stops playing, her career suddenly in jeopardy—the kind of jeopardy her father, Conrad, has always predicted; the kind he warned her about when he urged her to make more practical choices with her life.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B096HF9RJ4',
    title: 'All My Rage',
    authors: ['Sabaa Tahir'],
    averageRating: 4.7,
    description:
      'Misbah is a dreamer and storyteller, newly married to Toufiq in an arranged match. After their young life is shaken by tragedy, they come to the United States and open the Clouds Rest Inn Motel, hoping for a new start. ',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B092T8G139',
    title: 'Never Simple',
    authors: ['Liz Scheier'],
    averageRating: 4.3,
    description:
      'Scheier’s mother Judith was a news junkie, a hilarious storyteller, a fast-talking charmer you couldn’t look away from, a single mother whose devotion crossed the line into obsession, and—when in the grips of the mental illness that plagued every day of her life—a violent and abusive liar whose hold on reality was shaky at best. On an uneventful afternoon when Scheier was eighteen, her mother sauntered into the room to tell her two important things: one, she had been married for most of Scheier’s life to a man she’d never heard of, and two, the man she’d told Scheier was her father was entirely fictional.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B0927CQNWV',
    title: 'The Chase',
    authors: ['Candice Fox'],
    averageRating: 4.2,
    description:
      'With that, the largest manhunt in United States history is on. In response to a hostage situation, more than 600 inmates from the Pronghorn Correctional Facility, including everyone on Death Row, are released into the Nevada Desert. Criminals considered the worst of the worst, monsters with dark, violent pasts, are getting farther away by the second.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B096XGDWSX',
    title: 'In Love',
    authors: ['Amy Bloom'],
    averageRating: 4.6,
    description:
      'Amy Bloom began to notice changes in her husband, Brian: He retired early from a new job he loved; he withdrew from close friendships; he talked mostly about the past. Suddenly, it seemed there was a glass wall between them, and their long walks and talks stopped. Their world was altered forever when an MRI confirmed what they could no longer ignore: Brian had Alzheimer’s disease.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
  {
    id: 'B097B2DXS8',
    title: 'Disorientation',
    authors: ['Elaine Hsieh Chou'],
    averageRating: 4,
    description:
      'Twenty-nine-year-old PhD student Ingrid Yang is desperate to finish her dissertation on the late canonical poet Xiao-Wen Chou and never read about “Chinese-y” things again. But after years of grueling research, all she has to show for her efforts are junk food addiction and stomach pain. When she accidentally stumbles upon a curious note in the Chou archives one afternoon, she convinces herself it’s her ticket out of academic hell.',
    categories: ['detevite', 'adventure'],
    language: 'en',
    pageCount: 320,
    printType: 'test',
    publishedDate: 'test-date',
    publisher: 'test-publisher',
    ratingsCount: 3,
    subtitle: 'test-subtitle',
    imageLinks: { smallThumbnail: 'test', thumbnail: 'test' },
  },
];
