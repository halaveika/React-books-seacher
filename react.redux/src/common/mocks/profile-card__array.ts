export const mockCardArr = [
  {
    id: 'test1-id',
    firstname: 'test1-firstname',
    lastname: 'test1-firstname',
    zipcode: '22555',
    birthday: '01/01/2001',
    gender: 'test1-gender',
    avatar: 'test1-avatar',
    city: 'test1-city',
    gifts: ['test1-gift1', 'test1-gift2'],
  },
  {
    id: 'test2-id',
    firstname: 'test2-firstname',
    lastname: 'test2-firstname',
    zipcode: '22555',
    birthday: '02/02/2002',
    gender: 'test2-gender',
    avatar: 'test2-avatar',
    city: 'test2-city',
    gifts: ['test2-gift1', 'test2-gift2'],
  },
  {
    id: 'test3-id',
    firstname: 'test3-firstname',
    lastname: 'test3-firstname',
    zipcode: '22555',
    birthday: '03/03/2003',
    gender: 'test3-gender',
    avatar: 'test3-avatar',
    city: 'test3-city',
    gifts: ['test3-gift1', 'test3-gift2'],
  },
];
