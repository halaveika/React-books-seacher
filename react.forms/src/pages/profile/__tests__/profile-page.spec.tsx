import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ProfilePage from '../profile-page';
Enzyme.configure({ adapter: new Adapter() });
import { mockCardArr } from '../../../common/mocks/profile-card__array';

describe('test ProfilePage component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render ProfilePage component', async () => {
    const wrapper = shallow<ProfilePage>(<ProfilePage title="test" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should save value in localstorage during componentWillUnmount', () => {
    const wrapper = shallow<ProfilePage>(<ProfilePage title="test" />);
    jest.spyOn(Storage.prototype, 'setItem');
    const componentWillUnmount = jest.spyOn(wrapper.instance(), 'componentWillUnmount');
    wrapper.unmount();
    expect(componentWillUnmount.mock.calls.length).toBe(1);
  });

  it('should fire addCard', () => {
    const wrapper = shallow<ProfilePage>(<ProfilePage title="test" />);
    wrapper.instance().addCard(mockCardArr[0]);
    expect(wrapper.instance().state.cards).toEqual([mockCardArr[0]]);
  });
});
