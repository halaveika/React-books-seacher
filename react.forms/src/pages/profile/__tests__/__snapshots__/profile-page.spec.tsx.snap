// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ProfilePage component should render ProfilePage component 1`] = `
<Fragment>
  <Header
    title="test"
  />
  <main
    className="main container"
  >
    <div
      className="content"
    >
      <ProfileForm
        addCard={[Function]}
      />
      <ProfileCardList
        cards={Array []}
      />
    </div>
  </main>
</Fragment>
`;
